include/af/algorithm.h

```diff
@@ -1095,246 +1139,235 @@ extern "C" {
-
-       \note \p imag is always set to 0.
+       C Interface to check if all values over all dimensions are true.
+ 
+       \param[out] out 1 if all true; 0 otherwise
+       \param[in]  in  input array
+       \return         \ref af_err
 
        \ingroup reduce_func_all_true
     */
     AFAPI af_err af_all_true_all_array(af_array *out, const af_array in);
 #endif
 
     /**
-       C Interface for checking if any values in an array are true
-
-       \param[out] real is 1 if any value of input \p in is true, 0 otherwise.
-       \param[out] imag is always set to 0.
-       \param[in] in is the input array
-       \return \ref AF_SUCCESS if the execution completes properly
+       C Interface to check if any values over all dimensions are true.
 
-       \note \p imag is always set to 0.
+       \param[out] real 1 if any true; 0 otherwise
+       \param[out] imag 0
+       \param[in]  in   input array
+       \return          \ref af_err
 
        \ingroup reduce_func_any_true
     */
     AFAPI af_err af_any_true_all(double *real, double *imag, const af_array in);
 
 #if AF_API_VERSION >= 39
     /**
-       C Interface for checking if any values in an array are true,
-       while returning an af_array
+       C Interface to check if any values over all dimensions are true.
 
-       \param[out] out will contain 1 if any value of input \p in is true, 0 otherwise
-       \param[in] in is the input array
-       \return \ref AF_SUCCESS if the execution completes properly
-
-       \note \p imag is always set to 0.
+       \param[out] out 1 if any true; 0 otherwise
+       \param[in]  in  input array
+       \return         \ref af_err
 
        \ingroup reduce_func_any_true
     */
     AFAPI af_err af_any_true_all_array(af_array *out, const af_array in);
 #endif
 
     /**
-       C Interface for counting total number of non-zero values in an array
-
-       \param[out] real will contain the number of non-zero values in \p in.
-       \param[out] imag is always set to 0.
-       \param[in] in is the input array
-       \return \ref AF_SUCCESS if the execution completes properly
+       C Interface to count non-zero values over all dimensions.
 
-       \note \p imag is always set to 0.
+       \param[out] real count
+       \param[out] imag 0
+       \param[in]  in   input array
+       \return          \ref af_err
 
        \ingroup reduce_func_count
     */
     AFAPI af_err af_count_all(double *real, double *imag, const af_array in);
 
 #if AF_API_VERSION >= 39
     /**
-       C Interface for counting total number of non-zero values in an array,
-       while returning an af_array
+       C Interface to count non-zero values over all dimensions.
 
-       \param[out] out contain the number of non-zero values in \p in.
-       \param[in] in is the input array
-       \return \ref AF_SUCCESS if the execution completes properly
+       \param[out] out count
+       \param[in]  in  input array
+       \return         \ref af_err
```

Comment:
Let's tweak everywhere. ex:
af_err, will be AF_SUCCESS if function executes properly

Reply:
fixed